src/config.rs

```diff
@@ -107,6 +112,42 @@ impl Config {
             .build()?
             .try_deserialize()
     }
+    /// Retrieves all configured tokens.
+    #[allow(deprecated)]
+    pub fn get_tokens(&self) -> Option<Vec<String>> {
+        if self.server.auth_token.is_some() || self.server.auth_tokens.is_some() {
+            let mut merged: Vec<String> = vec![];
+            if let Some(tokens) = &self.server.auth_tokens {
+                merged = tokens.clone();
+            }
+            if let Some(token) = &self.server.auth_token {
+                log::warn!("[server].auth_token is deprecated, please use [server].auth_tokens");
+                merged.insert(0, token.to_string());
+            }
+            if let Ok(env_token) = env::var(AUTH_TOKEN_ENV) {
+                merged.insert(0, env_token);
+            }
+            if !merged.is_empty() {
+                return Some(merged);
+            }
+        }
+        None
+    }
```

Comment:
This function can be refactored as:

```rs
    pub fn get_tokens(&self) -> Option<Vec<String>> {
        let mut tokens = self.server.auth_tokens.clone().unwrap_or_default();
        if let Some(token) = &self.server.auth_token {
            log::warn!("[server].auth_token is deprecated, please use [server].auth_tokens");
            tokens.insert(0, token.to_string());
        }
        if let Ok(env_token) = env::var(AUTH_TOKEN_ENV) {
            tokens.insert(0, env_token);
        }
        (!tokens.is_empty()).then_some(tokens)
    }
```

Reply:
Wow, I think I still have to get more into the spirit of Rust.